### PR TITLE
add favorites list items

### PR DESCRIPTION
**Pull request by axce1**

### CODE REVIEW

File: dcf/static/dcf/css/style.css [hunk 1]
```diff
@@ -3,4 +3,9 @@
     position:relative;
     top: -20px;
     left: -15px;
+}
+
+span.glyphicon {
```
Comment (inoks) [inline dcf/static/dcf/css/style.css]: That's not good, since that affect all glyphicon site wide.

File: dcf/static/dcf/js/main.js [hunk 2]
```diff
@@ -3,26 +3,21 @@
  */
 jQuery(function($){
 
-  $("img.favorit").on("click", function(){
+  $("span.glyphicon").on("click", function(){
```
Comment (inoks) [inline dcf/static/dcf/js/main.js]: That's not good, since that affect all glyphicon site wide.

File: dcf/static/dcf/js/main.js [hunk 3]
```diff
@@ -3,26 +3,21 @@
...
             success: function() {
-                alert('good');
             }
           });
-
-
-
       } else {
-          $('img.favorit').attr('src', '/static/dcf/img/icon-check-unmark.png');
+          $("span.glyphicon").toggleClass("glyphicon-star glyphicon-star-empty")
```
Comment (inoks) [inline dcf/static/dcf/js/main.js]: That's not good, since that affect all glyphicon site wide.

File: dcf/static/dcf/js/main.js [hunk 4]
```diff
@@ -31,7 +26,6 @@ jQuery(function($){
             dataType: "json",
             data: { "item": document.getElementById("for_mark").value },
             success: function() {
```
Comment (inoks) [inline dcf/static/dcf/js/main.js]: No need to have empty functions

File: dcf/templates/_base.html [hunk 5]
```diff
@@ -24,6 +24,8 @@
         <link href="{% static 'dcf/css/lightbox.css' %}" rel="stylesheet" type="text/css"/>
         <link href="{% static 'dcf/css/style.css' %}" rel="stylesheet" type="text/css"/>
 
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (inoks) [inline dcf/templates/_base.html]: We already have bootstrap.css at the bottom of the page (which btw, should be moved to the header).

File: dcf/templates/dcf/item_list.inc.html [hunk 6]
```diff
@@ -24,7 +24,7 @@
                             {% endthumbnail %}
                         {% endif %}
                         <a href='{{ object.get_absolute_url }}' title='{{object.get_title}}'>
-                            {{object.title}}
+                          {{object.title}}
```
Comment (inoks) [inline dcf/templates/dcf/item_list.inc.html]: missing space after and before `{{` `}}`

File: dcf/views.py [hunk 7]
```diff
@@ -267,30 +268,25 @@ def page500(request):
...
+    user = request.user
     if request.is_ajax():
-        message = "Yes, AJAX!"
         user.favorites.add(item_id)
-        print(message)
+        message = "success"
     else:
-        message = "Not Ajax"
+        message = "error"
     return HttpResponse(message)
```
Comment (inoks) [inline dcf/views.py]: Should use `JsonResponse` instead

File: dcf/models.py [hunk 8]
```diff
@@ -115,3 +115,12 @@ def save(self, *args, **kwargs):
...
     file = ImageField(_('image'), upload_to='images')
+
+
+class FavoritsInfo(models.Model):
+    customuser = models.ForeignKey(CustomUser)
+    item = models.ForeignKey(Item)
+    liked = models.DateTimeField(null=True, blank=True, auto_now_add=True)
+
+    class Meta:
+        auto_created = True
```
Comment (inoks) [inline dcf/models.py]: I'm sure that Meta option works not as you expected. You need to have `DateTimeField` with `auto_now_add=True` option instead.

File: dcf/static/dcf/js/main.js [hunk 9]
```diff
@@ -3,29 +3,31 @@
...
           this.id = 'unmark';
 
           $.ajax({
             type: "POST",
             url: "/del_favorites/",
             dataType: "json",
             data: { "item": document.getElementById("for_mark").value },
-            success: function() {
+            success: function(data) {
+                data.msg
```
Comment (axce1) [inline dcf/static/dcf/js/main.js]: look at be6d4b5 commit

File: dcf/models.py (hunk 8 above)
Comment (axce1) [inline dcf/models.py]: fix model of last commit

**Comment by inoks** [comment]
Please squash your latests commits for that PR into one.

**Comment by axce1** [comment]
create new PR

**Pull request closed without merging**